app/views/spree/users/show.html.haml

```diff
@@ -1,5 +1,5 @@
 - content_for :injection_data do
-  = inject_json_array("orders", @orders.all, Api::OrderSerializer)
+  = inject_json_array("orders", @orders.all.includes(:line_items), Api::OrderSerializer)
```

Comment:
Nice :100:

I wonder if there are any other associated objects that would benefit from eager-loading here? :thinking:

Just out of interest; were you using rack-mini-profiler to inspect the queries?

Reply:
I did use mini-profiler. It was super easy to just CTRL+F :smile:

Reply:
Should the `includes` part be moved from the view, to the controller?

https://github.com/openfoodfoundation/openfoodnetwork/blob/7f27544acb5bb4d4f3dc2dfd01848abe2046dddb/app/controllers/spree/users_controller.rb#L15-L18

https://github.com/openfoodfoundation/openfoodnetwork/blob/f49d4592a0d42783b712c571856e6e9a3eaf27d3/app/controllers/spree/users_controller.rb#L62-L64

Reply:
Yeah I think it'd be best practice. So I just updated it. Good catch. Thanks! :smile:

Reply:
good move guys! Now I see we totally missed the `includes` was being called on `all`, which returns an array :see_no_evil: :thinking: In any case, separating the view's and controller's concerns will make things maintainable.

Reply:
@sauloperez Where do you think it make sense to have the `includes`?

Is this where the final query is assembled?

https://github.com/openfoodfoundation/openfoodnetwork/blob/f49d4592a0d42783b712c571856e6e9a3eaf27d3/app/queries/complete_orders_with_balance.rb#L15-L20

Would it make sense to have the includes here, or is this query used in other places where we don't want `line_item` included?

Reply:
@viktorsmari, @sauloperez  So I just tried to scour the code base to see if `CompleteOrdersWithBalance` was used anywhere else, and I really couldn't find it. Would doing it like I did in the example below make any real world difference?

This is how the output looks like when you eager load from `sorted_finalized_orders` method : 

```zsh
web_1     |   Spree::LineItem Load (4.1ms)  SELECT "spree_line_items".* FROM "spree_line_items" WHERE "spree_line_items"."order_id" IN ($1, $2, $3, $4, $5) ORDER BY created_at ASC  [["order_id", 10], ["order_id", 9], ["order_id", 8], ["order_id", 7], ["order_id", 6]]
web_1     |   Enterprise Load (2.8ms)  SELECT  "enterprises".* FROM "enterprises" WHERE "enterprises"."id" = $1 LIMIT $2  [["id", 6], ["LIMIT", 1]]
web_1     |   Spree::Payment Load (2.2ms)  SELECT "spree_payments".* FROM "spree_payments" INNER JOIN "spree_payment_methods" ON "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."id" = "spree_payments"."payment_method_id" WHERE "spree_payments"."order_id" = $1 AND (state IN ('completed','pending'))  [["order_id", 10]]
web_1     |   Enterprise Load (1.6ms)  SELECT  "enterprises".* FROM "enterprises" WHERE "enterprises"."id" = $1 LIMIT $2  [["id", 4], ["LIMIT", 1]]
web_1     |   Spree::Payment Load (2.3ms)  SELECT "spree_payments".* FROM "spree_payments" INNER JOIN "spree_payment_methods" ON "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."id" = "spree_payments"."payment_method_id" WHERE "spree_payments"."order_id" = $1 AND (state IN ('completed','pending'))  [["order_id", 9]]
web_1     |   Enterprise Load (1.6ms)  SELECT  "enterprises".* FROM "enterprises" WHERE "enterprises"."id" = $1 LIMIT $2  [["id", 3], ["LIMIT", 1]]
web_1     |   Spree::Payment Load (2.2ms)  SELECT "spree_payments".* FROM "spree_payments" INNER JOIN "spree_payment_methods" ON "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."id" = "spree_payments"."payment_method_id" WHERE "spree_payments"."order_id" = $1 AND (state IN ('completed','pending'))  [["order_id", 8]]
web_1     |   CACHE Enterprise Load (0.1ms)  SELECT  "enterprises".* FROM "enterprises" WHERE "enterprises"."id" = $1 LIMIT $2  [["id", 3], ["LIMIT", 1]]
web_1     |   Spree::Payment Load (2.2ms)  SELECT "spree_payments".* FROM "spree_payments" INNER JOIN "spree_payment_methods" ON "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."id" = "spree_payments"."payment_method_id" WHERE "spree_payments"."order_id" = $1 AND (state IN ('completed','pending'))  [["order_id", 7]]
web_1     |   Enterprise Load (2.1ms)  SELECT  "enterprises".* FROM "enterprises" WHERE "enterprises"."id" = $1 LIMIT $2  [["id", 5], ["LIMIT", 1]]
web_1     |   Spree::Payment Load (1.8ms)  SELECT "spree_payments".* FROM "spree_payments" INNER JOIN "spree_payment_methods" ON "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."deleted_at" IS NULL AND "spree_payment_methods"."id" = "spree_payments"."payment_method_id" WHERE "spree_payments"."order_id" = $1 AND (state IN ('completed','pending'))  [["order_id", 6]]
```

Reply:
`CompleteOrdersWithBalance` is meant to be chainable with other ActiveRecord relations and as such, it can be used in different contexts. Therefore, any includes a particular context may require, need to be added by the caller and not this query object.